assets/quick-add-bulk.js

```diff
@@ -24,7 +24,10 @@ if (!customElements.get('quick-add-bulk')) {
 
       connectedCallback() {
         this.cartUpdateUnsubscriber = subscribe(PUB_SUB_EVENTS.cartUpdate, (event) => {
-          if (event.source === 'quick-add') {
+          if (
+            event.source === 'quick-add' ||
+            (event.cartData.items && event.cartData.items.find((item) => item.id === this.dataset.index) === undefined)
```

Comment:
```suggestion
            (event.cartData.items && !event.cartData.items.some((item) => item.id === this.dataset.index))
```
This seems like a good use case for `some` since you are just looking for a true/false statement in the end: "Are there any variant IDs in the cart that match this one?"

With that said: I was never able to create a scenario in the bulk-add modal where this line wasn't true. I was always getting an early return. How come you needed to add this line?

You wrote:
> Ensuring that the pubsub only updates if the variant id matches to avoid multiple calls and failed calls since the id doesnt exist

What is the steps to do to fire this function, but not encounter an early return?